integration/admin-client/src/main/java/org/keycloak/admin/client/resource/GroupsResource.java

```diff
@@ -124,4 +124,7 @@ List<GroupRepresentation> groups(@QueryParam("search") String search,
     @Path("{id}")
     GroupResource group(@PathParam("id") String id);
 
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    List<GroupRepresentation> query(@QueryParam("q") String searchQuery);
```

Comment:
Strictly speaking, we only need this feature in the SPI, so if it helps to reduce scope we can drop HTTP API from the PR. Needs maintainer feedback.

Reply:
As we referred the "search client by attributes" PRs, we belive there shouldn't be any major review comments as part of this. If we have to do any major refactoring in API side, we will drop API changes